templates/todo/projects/csharp-cosmos-sql/.repo/bicep/infra/main.bicep

```diff
@@ -141,6 +144,40 @@ module monitoring '../../../../../../common/infra/bicep/core/monitor/monitoring.
     logAnalyticsName: !empty(logAnalyticsName) ? logAnalyticsName : '${abbrs.operationalInsightsWorkspaces}${resourceToken}'
     applicationInsightsName: !empty(applicationInsightsName) ? applicationInsightsName : '${abbrs.insightsComponents}${resourceToken}'
     applicationInsightsDashboardName: !empty(applicationInsightsDashboardName) ? applicationInsightsDashboardName : '${abbrs.portalDashboards}${resourceToken}'
+    useAPIM: useAPIM
+  }
+}
+
+resource appInsightsService 'Microsoft.ApiManagement/service@2021-08-01' existing = {
+  name: monitoring.outputs.applicationInsightsName
+  scope: rg
+}
+
+// Creates Azure API Management (APIM) service to mediate the requests between the frontend and the backend API
+module apim '../../../../../../common/infra/bicep/core/host/apim.bicep' = if (useAPIM) {
+  name: 'apim-deployment'
+  scope: rg
+  params: {
+    name: 'apim-${resourceToken}'
```

Comment:
We shouldn't need resourceToken here.

Reply:
This is to provide a unique name for the service

Reply:
Like this: !empty(apimServiceName) ? apimServiceName : '${abbrs.apimService}${resourceToken}'

Reply:
Yes